tests/validation/tests/v3_api/test_monitoring.py

```diff
@@ -0,0 +1,299 @@
+    wait_for_target_up("expose-prometheus-metrics")
+    wait_for_target_up("expose-grafana-metrics")
+
+
+def test_monitoring_workload_metrics():
+    _, cluster = get_admin_client_and_cluster()
+    project = namespace["project"]
+    project_client = get_project_client_for_token(project, ADMIN_TOKEN)
+    # deploy a workload to test project monitoring
+    cluster_client = get_cluster_client_for_token(cluster, ADMIN_TOKEN)
+    ns = create_ns(cluster_client, cluster, project, random_name())
+    port = {"containerPort": 8080,
+            "type": "containerPort",
+            "kind": "NodePort",
+            "protocol": "TCP"}
+    workloadMetrics = [{"path": "/metrics",
+                        "port": 8080,
+                        "schema": "HTTP"}]
+    con = [{"name": "test-web",
+            "image": "loganhz/web",
+            "ports": [port]}]
+    wl_name = random_name()
+    workload = project_client.create_workload(name=wl_name,
+                                              containers=con,
+                                              namespaceId=ns.id,
+                                              workloadMetrics=workloadMetrics)
+    app = project_client.list_app(name=PROJECT_MONITORING_APP).data[0]
+    url = CATTLE_TEST_URL + '/k8s/clusters/' + cluster.id + '/api/v1/namespaces/' + app.targetNamespace \
+          + '/services/http:access-prometheus:80/proxy/api/v1/query?query=web_app_online_user_count'
+    headers1 = {'Authorization': 'Bearer ' + ADMIN_TOKEN}
+    start = time.time()
+    while True:
+        result = requests.get(headers=headers1, url=url, verify=False).json()
+        if len(result["data"]["result"]) > 0:
+            return
+        if time.time() - start > DEFAULT_MULTI_CLUSTER_APP_TIMEOUT:
+            raise AssertionError(
+                "Timed out waiting for the graph data is available in Prometheus")
+        time.sleep(5)
+
+
+@pytest.fixture(scope="module", autouse="True")
+def create_project_client(request):
+    rancher_client, cluster = get_admin_client_and_cluster()
+    create_kubeconfig(cluster)
+    project = create_project(rancher_client, cluster,
+                             random_test_name("p-monitoring"))
+    namespace["cluster"] = cluster
+    namespace["project"] = project
+
+    def fin():
+        rancher_client.delete(project)
+        cluster2 = rancher_client.reload(cluster)
+        if "disableMonitoring" in cluster2.actions:
+            cluster2.disableMonitoring()
+
+    request.addfinalizer(fin)
+
+
+def get_graph_data(query1, list_len, timeout=DEFAULT_MULTI_CLUSTER_APP_TIMEOUT):
+    rancher_client, cluster = get_admin_client_and_cluster()
+    start = time.time()
+    while True:
+        graphs = rancher_client.action(**query1)
+        if hasattr(graphs, "data") is True:
+            print("the length is ", len(graphs.data))
+            print(graphs.data)
+            if len(graphs.data) == list_len:
+                return graphs.data
```

Comment:
@jiaqiluo  Can we also additionally check that we get a non zero data for every metrics in the graph data ?

Reply:
fixed